unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -39,10 +40,14 @@ public class PhysicsRemoteFPSAgentController : BaseFPSAgentController {
 
         [SerializeField] protected bool IsHandDefault = true;
 
+        //DRONE parameters
         [SerializeField] public bool FlightMode = false;
+        public bool hasFixedUpdateHappened = true;//track if the fixed physics update has happened
         protected Vector3 thrust;
-
-        [SerializeField] Camera[] FlightCameras;
+        public float dronePositionRandomNoiseSigma = 0f;
+        //count of fixed updates for use in droneCurrentTime
+        public float fixupdateCnt = 0f;
+        public float autoResetTimeScale = 1.0f;
```

Comment:
I think these variables should be in a controller that inherits from `PhysicsRemoteFPSAgentController` just like the stochastic controller, something like `DroneAgentController`. Any sub class now has `dronePositionRandomNoiseSigma` which is not used, and drone actions should throw unsupported exceptions for the other controllers

Reply:
agreed, its messy enough as it is, and this might make it a bit more organized at the very least.